src/component/sign-in/ui-input/Input.tsx

```diff
@@ -0,0 +1,35 @@
+import { forwardRef } from 'react';
+import { Noop } from 'react-hook-form';
+import classNames from 'classnames/bind';
+import styles from './Input.module.css';
+
+let cx = classNames.bind(styles);
+
+export interface InputProps {
+  placeholder: string;
+  type?: string;
```

Comment:
조금 더 안전하게 하려면, input이 받을 수 있는 모든 type의 리터럴 타입을 유니온으로 미리 선언해주면 사용하는 사람 입장에서는 편하겠죠?
그대신 만든사람입장에서는 귀찮은 일이에요.
그래서 만드는 사람과 사용하는 사람의 편의를 잘 고려해야해요

Reply:
```suggestion
type InputTypes = "text" | "password" | "button" | "checkbox" | "color" | "email" 
//...
  type: InputTypes
```
위와 같이 말씀하시는 걸까요? 오오 사용할때 훨씬 편할 것 같습니다.
이때는 앞으로의 컴포넌트 활용성을 위해 모든 input type 종류를 넣는 것이 좋을까요, 
아니면 현재 사용되는 타입("text", "password")만 넣으면 좋을까요?

제가 고민해봤을 때, 사용하는 사람 입장에서 type이 2가지로 제한되어 있으면 다른 타입 사용이 불가하다고 판단할 수도 있을 것 같아서 모든 타입을 작성해주는 것이 좋을 것 같다는 생각이 듭니다.

Reply:
서영님 생각도 맞아요. 저라면 일단 text,password 외의 타입은 아직 커스텀 하게 정의한게 없으니,
type InputTypes = "text" | "password"
해놓고 위에 주석으로 추가적인 타입은 구현 후 InputTypes에 추가 필요
이런식으로 관리하면 다른사람도 충분히 이해할 수 있을거라 생각해요